src/lib.rs

```diff
@@ -420,10 +416,42 @@ macro_rules! derive_alias {(
                 )}
                 #[allow(unused_imports)]
                 pub(in crate) use [< $MacroName __derive_macro >] as $MacroName;
-            )*
-        }
-    )}
-)}
+            }
+        )}
+    };
+    (pub #[derive($MacroName:ident !)] = #[derive($($derives:tt)*)];) => {
+        $crate::ඞ_with_dollar! {( $_:tt ) => (
+            $crate::ඞ::paste! {
+                // To avoid ambiguities with what the re-export
+                // refers to, let's use a hopefully unused name.
+                //
+                // Indeed, eponymous derive macros in scope such as those
+                // from the prelude would otherwise cause trouble with the
+                // re-export line.
+                #[allow(nonstandard_style)]
+                #[macro_export]
+                macro_rules! [< $MacroName __derive_macro >] {(
+                    $_($item:tt)*
+                ) => (
+                    $crate::ඞ_nested_derive! {
+                        #[derive($($derives)*)]
+                        $_($item)*
+                    }
+                )}
+                #[allow(unused_imports)]
+                pub use [< $MacroName __derive_macro >] as $MacroName;
+            }
+        )}
+    };
```

Comment:
```suggestion
    );
```

Macros map input tokens to output tokens, and these sequences of tokens are `()`-delimited in input, so I `()`-delimit them in output too; usage of `{}` is not a great habit, since it can lead people to forget actual surrounding `{}` when needed (it won't play nicely with `rustfmt`, but at the same time, `rustfmt` is not great on macro definitions anyways).

  - If you could adjust the other macro rules accordingly it would be great 🙏